syntax = "proto3";

import "google/protobuf/timestamp.proto";
option java_package = "com.userinfoservice.grpc";

message Date {
	int32 year = 1;
	int32 month = 2;
	int32 day = 3;
}

message SetUserInfoRequest {
	string first_name = 1;
	string last_name = 2;
	Date date_of_birth = 3;
	string email = 4;
	string phone_number = 5;
	google.protobuf.Timestamp timestamp = 6;
}

message SetUserInfoResponse { }

service UserInfo {
	rpc SetUserInfo(SetUserInfoRequest) returns (SetUserInfoResponse) { }
}